### PR TITLE
tests: add multipart tests, rearrange testing scripts

### PR DESCRIPTION
**Pull request by jecluis**
# Describe your changes
This patchset introduces multipart upload tests, rearranges python test names, and introduces a top script to run all tests.

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

File: tests/test-s3gw-multipart.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+        self.assertTrue(down_md5.hexdigest() == md5)
+
+    def test_upload_multipart_manual(self):
+        bucket_name = self.get_random_bucket_name()
+        self.s3c.create_bucket(Bucket=bucket_name)
+        self.assert_bucket_exists(bucket_name)
+
+        objname = self.get_random_object_name()
+        objsize = 100 * 1024**2  # 10MB
```
Comment (giubacc) [inline tests/test-s3gw-multipart.py:188]: wrong comment? is this 100 mb?

File: tests/test-s3gw-multipart.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+        partsize = 10 * 1024**2  # 1MB
```
Comment (giubacc) [inline tests/test-s3gw-multipart.py:189]: wrong comment? is this 10 mb?

File: tests/test-s3gw-multipart.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+            print(f"actual md5: {md5_2.hexdigest()}")
+        self.assertTrue(md5.hexdigest() == mp.md5)
+
+    def test_list_ongoing_parts(self):
+        bucket_name = self.get_random_bucket_name()
+        self.s3c.create_bucket(Bucket=bucket_name)
+        self.assert_bucket_exists(bucket_name)
+
+        objname = self.get_random_object_name()
+        objsize = 100 * 1024**2  # 10MB
```
Comment (giubacc) [inline tests/test-s3gw-multipart.py:253]: check these comments also

File: tests/test-s3gw-multipart.py (hunk 2 above)
Comment (jecluis) [reply tests/test-s3gw-multipart.py:189]: yeah, forgot to change the comment :)

**Comment by 0xavi0** [comment]
Just a warning here.

The `REST API` tests are broken (at least when running with the multipart PR's code https://github.com/aquarist-labs/ceph/pull/40

For example:
```bash
python test-s3gw-users-rest-api.py 127.0.0.1:7480
F
======================================================================
FAIL: test_smoke_test (__main__.UserRestAPITests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test-s3gw-users-rest-api.py", line 65, in test_smoke_test
    self.assertEqual(200, response.status_code)
AssertionError: 200 != 409

----------------------------------------------------------------------
Ran 1 test in 0.037s

FAILED (failures=1)
```
That line is removing a user, so I guess something broke that feature (I'll investigate and open a bug if needed)


The versioning tests will also be broken because of a `etag` bug fix done in the multipart PR (they also need to be rebased to get the latests changes for versioning and delete markers)

I already have a fixed version of the versioning tests ready for pushing when the multipart PR is merged.

**Comment by jecluis** [comment]
linter failure is silly

**Comment by jecluis** [comment]
linter failure has been addressed by changing the way the linter checks for duplicate headers. Otherwise we can't have subheaders that are the same. :)

**Comment by jecluis** [comment]
Any other thoughts before merging this?

**Review by giubacc — APPROVED** [review]
it sounds good to me

**Pull request merged**